src/Core/PluginInterface.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright 2005-2011, Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Cake\Core;
+
+/**
+ * Plugin Interface
+ */
+interface PluginInterface
+{
+    /**
+     * Get the name of this plugin.
+     *
+     * @return void
+     */
+    public function getName();
+
+    /**
+     * Get the filesystem path to this plugin
+     *
+     * @return void
+     */
+    public function getPath();
+
+    /**
+     * Get the filesystem path to configuration for this plugin
+     *
+     * @return void
+     */
+    public function getConfigPath();
+
+    /**
+     * Get the filesystem path to configuration for this plugin
+     *
+     * @return void
+     */
+    public function getClassPath();
+
+    /**
+     * Load all the application configuration and bootstrap logic.
+     *
+     * Override this method to add additional bootstrap logic for your application.
+     *
+     * @return void
+     */
+    public function bootstrap();
```

Comment:
Should this interface define other hooks (`console` etc) as well? 

Or maybe it should be split into `ConsolePlugin` and `HttpPlugin` interfaces? I'm wondering if we should keep different interfaces for plugins and regular app. App is nicely  defined through interfaces (console, http, upcoming events) but here we just throw everything into one bucket.

Reply:
What benefits are there in having 3 interfaces for plugins? With the application interfaces it enabled standalone console applications to not concern themselves with http. This may also have been a side-effect of the `BaseApplication` being in the Http namespace. With plugins the base class is in Core along with the interface.